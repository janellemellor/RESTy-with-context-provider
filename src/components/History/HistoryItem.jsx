import React from 'react';
import PropTypes from 'prop-types';

const HistoryItem = ({ url, method }) => (
  <section>
    <p>{method}</p>
    <p>{url}</p>
  </section>
); 

HistoryItem.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired
};

export default HistoryItem;
