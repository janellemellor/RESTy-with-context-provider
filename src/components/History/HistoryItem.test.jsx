import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe('HistoryItem Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<HistoryItem method='GET' url='https://github.com/janellemellor' />);
    expect(wrapper).toMatchSnapshot();
  });  
});

