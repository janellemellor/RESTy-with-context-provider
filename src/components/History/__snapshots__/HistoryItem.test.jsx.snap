// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem Component matches a snapshot 1`] = `
<section>
  <p>
    GET
  </p>
  <p>
    https://github.com/janellemellor
  </p>
</section>
`;
